### ISSUE TITLE
rocjitsu (gfx1250): SIGSEGV in L2Cache::send_backing on a GEMM/matmul with large output dim (vector-load path)

### ISSUE THREAD

**Issue by vivekkhandelwal1** [ROCm/rocm-systems#8751]
## Summary

On an emulated **gfx1250 (MI450X)** device under **rocjitsu** (functional mode, via mirage), a single **matmul / GEMM with a large output dimension** crashes with a native **SIGSEGV in the emulated vector-load cache path**. A plain `torch.matmul` reproduces it — no model, optimizer, or attention involved — so torch training/inference on gfx1250 is currently blocked (embedding/vocab projections and typical hidden/FFN GEMMs cross the threshold).

This is a **different code path** from the recently fixed scalar-memory issue (#8539 / #8652, `s_load` ioffset scaling): that fix is present and works; this crash is on the **vector global-load** path.

## Environment

- rocjitsu functional emulator launched via `mirage run`, built from the current `emulation/mirage` (includes #8652).
- Emulated agent: MI450X (gfx1250); pure software emulation, **no physical GPU**.
- Container torch built for gfx1250 (`torch.cuda.get_arch_list() == ['gfx1250']`).

## Steps to reproduce

1. Build mirage and create a gfx1250 profile:

```bash
# from ROCm/rocm-systems
cd emulation/mirage
./scripts/mirage-docker-build.sh
export MIRAGE_BIN="$PWD/build/manylinux/bin/mirage"

"$MIRAGE_BIN" profile create mi450x \
  --emulator rocjitsu --agent MI450X \
  --num-nodes 1 --gpus-per-node 1 --no-input
```

2. Save `repro.py`:

```python
import torch

d = torch.device("cuda:0")
a = torch.randn(64, 128, device=d)
b = torch.randn(128, 1024, device=d)   # output dim 1024 -> crash; <= ~912 is fine
c = a @ b
torch.cuda.synchronize()
print("OK", tuple(c.shape))
```

3. Run under mirage, using a container image whose torch is built for gfx1250:

```bash
"$MIRAGE_BIN" run --in-process --profile mi450x \
  --image <ROCm image with a gfx1250 torch> \
  --mount "$PWD/repro.py:/repro.py:ro" \
  -- python3 /repro.py
```

**Observed:** `Segmentation fault (core dumped)` (exit 139).
**Expected:** prints `OK (64, 1024)`.

## Output-dimension threshold

Fixing `M=64, K=128` and sweeping the output dim `N` (same result for `torch.matmul`, `F.linear`, and `nn.Linear`):

| N | ≤ 912 | 960 | 1024 | 2048 |
|---|-------|-----|------|------|
| result | OK | crash | crash | crash |

The crash appears once `N` crosses ~912–960, i.e. when the GEMM library selects a different kernel variant. It is deterministic (fails every run in a fresh process).

## Native backtrace

`librocjitsu.so` (not stripped). The fault is on a simulation worker thread servicing a global **vector** load:

```
Thread received signal SIGSEGV, Segmentation fault.
=> mov (%rsi),%rax        # rsi = 0x1000, si_addr = 0x1000  (near-null deref)
#0  rocjitsu::amdgpu::L2Cache::send_backing(...)
#1  rocjitsu::amdgpu::L2Cache::ensure_line(...)
#2  rocjitsu::amdgpu::L2Cache::fetch_line(...)
#3  rocjitsu::amdgpu::L1VectorCache::ensure_line(...)
#4  rocjitsu::amdgpu::L1VectorCache::read_bytes(...)
#5  rocjitsu::amdgpu::L1VectorCache::load(...)
#6  rocjitsu::amdgpu::GlobalMemPipeline::initiate_access(...)
#7  rocjitsu::amdgpu::MemoryPipeline::issue(...)
#8  rocjitsu::amdgpu::ComputeUnitCore::issue_instruction(...)
#9  simdojo::SimulationEngine::worker_loop(...)  ->  rj_vm_run()
```

While filling a cache line for the GEMM's vector load, `send_backing` dereferences a near-null pointer (faulting address `0x1000`) — an unguarded out-of-range access in the emulated memory backing rather than a wild GPU address.

## Pointers

- `emulation/rocjitsu/lib/rocjitsu/src/rocjitsu/vm/amdgpu/l2_cache.cpp` — `L2Cache::send_backing` (crash frame).
- `emulation/rocjitsu/lib/rocjitsu/src/rocjitsu/isa/arch/amdgpu/gfx1250/addr_calc.cpp` — `flat_global_calculate_addresses` applies `vaddr *= scale` (`scaled_vaddr_factor`) under the same `scale_offset` folding that #8652 addressed for scalar loads; possibly a sibling issue in the vector path (unconfirmed).

## Notes

gfx1250 simulation is listed as Experimental in the rocjitsu support matrix. Filing separately from #8539 since that specific (scalar `s_load`) issue is fixed and verified, while this vector-load GEMM crash remains.

**Comment by kuhar** [comment]
@vivekkhandelwal1 Hi Vivek, can you include pytorch setup steps in your repro?

**Comment by vivekkhandelwal1** [comment]
I tested the patch and the issue is fixed now. Thanks!
